File: uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/OnChatInputChangeListener.java
package io.agora.chat.uikit.chat.interfaces;

public interface OnChatInputChangeListener {
    /**
     * EditText text change monitoring
     * @param s
     * @param start
     * @param before
     * @param count
     */
    void onTextChanged(CharSequence s, int start, int before, int count);
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/feature/chat/interfaces/OnChatInputChangeListener.kt
package com.hyphenate.easeui.feature.chat.interfaces

import android.text.Editable
import android.view.KeyEvent
import android.view.View

interface OnChatInputChangeListener {
    /**
     * EditText text change monitoring
     * @param s
     * @param start
     * @param before
     * @param count
     */
    fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int)

    /**
     * After typing on the editing text layout.
     */
    fun afterTextChanged(s: Editable?){}

    /**
     * Edit text layout key events.
     */
    fun editTextOnKeyListener(v: View?, keyCode: Int, event: KeyEvent?):Boolean
}